htdocs/class/cache/model.php

```diff
@@ -122,22 +122,22 @@ public function gc()
      * Write data for key into cache
      *
      * @param  string  $key      Identifier for the data
-     * @param  mixed   $data     Data to be cached
+     * @param  mixed   $value     Data to be cached
      * @param  integer $duration How long to cache the data, in seconds
      * @return boolean True if the data was successfully cached, false on failure
      * @access public
      */
-    public function write($key, $data, $duration)
+    public function write($key, $value = null, $duration = null)
```

Comment:
This is another upside down fix. \XoopsCacheEngine::write() should not allow the $value to be omitted. Since the writes to the engine should come from \XoopsCache::write() which doesn't allow a missing value, this shouldn't be a problem to change the signature in XoopsCacheEngine.